deps/gen_functions.c

```diff
@@ -57,6 +58,7 @@ int main(int argc, char *argv[])
   printf("const MPI_WAITALL            = Libdl.dlsym(libmpi, \"%s\")\n", STRING(MPI_WAITALL));
   printf("const MPI_WAITANY            = Libdl.dlsym(libmpi, \"%s\")\n", STRING(MPI_WAITANY));
   printf("const MPI_WTIME              = Libdl.dlsym(libmpi, \"%s\")\n", STRING(MPI_WTIME));
+  printf("const MPI_IN_PLACE           = Base.box(Ptr{Int}, Base.unbox(Int, Int(%d)))\n", MPI_IN_PLACE);
```

Comment:
This looks really weird.  `MPI_IN_PLACE` is an integer, but you are casting it to a pointer?  I don't think this is what you want.  In any case, probably we need the Fortran version of this constant, so it should be defined in `gen_constants.f`, not here.